routes/api.php

```diff
@@ -47,20 +49,23 @@
             Route::resource('/', "H5pController");
             Route::get('settings', "H5pController@create");
             Route::get('embed/{id}', "H5pController@embed");
+            Route::get('activity/{activity}', "H5pController@showByActivity");
             //H5P Ajax calls
-            //Route::get('ajax/libraries', '\Djoudi\LaravelH5p\Http\Controllers\AjaxController@libraries')->name("h5p.ajax.libraries");
             Route::match(['GET', 'POST'], 'ajax/libraries', '\Djoudi\LaravelH5p\Http\Controllers\AjaxController@libraries')->name("h5p.ajax.libraries");
             Route::get('ajax/single-libraries', '\Djoudi\LaravelH5p\Http\Controllers\AjaxController@singleLibrary')->name("h5p.ajax.single-libraries");
             Route::any('ajax/content-type-cache', '\Djoudi\LaravelH5p\Http\Controllers\AjaxController@contentTypeCache')->name("h5p.ajax.content-type-cache");
             Route::any('ajax/library-install', '\Djoudi\LaravelH5p\Http\Controllers\AjaxController@libraryInstall')->name("h5p.ajax.library-install");
             Route::post('ajax/library-upload', '\Djoudi\LaravelH5p\Http\Controllers\AjaxController@libraryUpload')->name("h5p.ajax.library-upload");
             Route::post('ajax/rebuild-cache', '\Djoudi\LaravelH5p\Http\Controllers\AjaxController@rebuildCache')->name("h5p.ajax.rebuild-cache");
-            Route::any('ajax/files', '\Djoudi\LaravelH5p\Http\Controllers\AjaxController@files')->name("h5p.ajax.files");
             Route::any('ajax/filter', '\Djoudi\LaravelH5p\Http\Controllers\AjaxController@filter')->name("h5p.ajax.filter");
             Route::any('ajax/finish', '\Djoudi\LaravelH5p\Http\Controllers\AjaxController@finish')->name("h5p.ajax.finish");
             Route::any('ajax/content-user-data', '\Djoudi\LaravelH5p\Http\Controllers\AjaxController@contentUserData')->name("h5p.ajax.content-user-data");
         });
     });
 
+    //H5P Activity public route
+    Route::get('h5p/activity/{activity}/{visibility}', "H5pController@showByActivity");
+    //Route to support H5P Editor's core js library fileupload with "new XMLHttpRequest()"   
+    Route::any('h5p/ajax/files', '\Djoudi\LaravelH5p\Http\Controllers\AjaxController@files')->name("h5p.ajax.files");
```

Comment:
And perhaps, this could be;  h5p/upload-files ? (@yurysakovich1010  - could you comment on this?)
Waqar mentioned that H5P JS file needs a public route for upload files from editor, and that we provide this route as config in djoudi codebase.

Reply:
As a reference for routes naming
https://github.com/djoudi/Laravel-H5P/blob/master/routes/laravel-h5p.php

again we will refactor whole H5P codebase in coming release. this was discussed

Reply:
Acknowledged, parked for now!